src/utils/google.js

```diff
@@ -0,0 +1,48 @@
+const passport = require('passport');
+const GoogleStrategy = require('passport-google-oidc');
+const { User, Volunteer } = require('../models/user');
+
+passport.use(
+  new GoogleStrategy(
+    {
+      clientID: process.env.GAPP_CLIENT,
+      clientSecret: process.env.GAPP_SECRET,
+      callbackURL: 'http://localhost:8000/auth/google/callback',
+      scope: ['profile', 'email', 'openid'],
+    },
+    async (_, profile, cb) => {
+      /* 
+    Check if a user with googleId or email exists. 
+    Create user if there is no user associated with credentials 
+    */
+      User.findOne(
+        {
+          $or: [{ googleId: profile.id }, { email: profile.emails[0].value }],
+        },
+        (err, user) => {
```

Comment:
Better to invoke the user model functions using async/await instead of callback functions and use try/catch to handle errors.